data/core/common.lua

```diff
@@ -273,4 +273,26 @@ function common.relative_path(ref_dir, dir)
 end
 
 
+function common.mkdirp(path)
+  local stat = system.get_file_info(path)
+  if stat and stat.type then
+    return false, "path exists", path
```

Comment:
by convention, lua error usually returns `nil` instead of `false`, but it's just a nitpick.

Reply:
Hmmm, I see. I didn't want to nitpick, I didn't know about this convention. Normally I try to respect the rule of keeping the same type for variables so if you returns true on success I return false or error and vice-versa.

I am okay with using nil but since this is cosmetic I will leave it as it is. We may fix these later to have a more coherent coding style.